commands/promote.js

```diff
@@ -61,6 +63,45 @@ function * run (context, heroku) {
       }
     })
   }))
+
+  let releasePhase = (yield heroku.get(`/apps/${app}/formation`))
+    .find((formation) => formation.type === 'release')
+
+  if (releasePhase) {
+    yield cli.action('Checking release phase', co(function * () {
+      let releases = (yield heroku.get(`/apps/${app}/releases`))
+          .sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at))
+      let attach = releases.find((release) => release.description.includes('Attach DATABASE'))
+      let detach = releases.find((release) => release.description.includes('Detach DATABASE'))
+
+      if (!attach || !detach) {
+        throw new Error(`Unable to find Detach and Attach DATABASE releases ${cli.color.app(app)}`)
```

Comment:
So in this case, the user would see `Promoting...`, then `Checking release phase`, and then possibly this error? This message doesn't really give context for what's going on here.

Reply:
Yeah this error message can be confusing to the customer. I'd say that we can just ignore this (we didn't have this check before so not showing anything is not degrading), maybe?

Reply:
changed the message to `throw new Error('Unable to check release phase. Check your Attach DATABASE release for failures.')`